projs/shadow/shadow-engine/inc/core/ModuleManager.h

```diff
@@ -9,7 +9,7 @@ namespace ShadowEngine {
 
     struct ModuleRef{
     public:
-        Module* module;
+        std::shared_ptr<Module> module;
```

Comment:
The module can't be a shared pointer cause for hot reloading we need to forcefully unload it.

Reply:
We can forcefully delete the shared pointer. However, we'd need to tell the dll itself to do it.